cibuildwheel/architecture.py

```diff
@@ -77,6 +81,26 @@ def all_archs(platform: PlatformName) -> 'Set[Architecture]':
         else:
             assert_never(platform)
 
+    @staticmethod
+    def bitiness_archs(platform: PlatformName, bitiness: Literal["64", "32"]) -> 'Set[Architecture]':
+        native_architecture = Architecture(platform_module.machine())
+
+        if native_architecture in {Architecture.x86_64, Architecture.aarch64, Architecture.ppc64le, Architecture.s390x, Architecture.AMD64}:
+            if bitiness == "64":
+                return {native_architecture}
+            else:
+                if native_architecture == Architecture.x86_64 and platform != "macos":
+                    return {Architecture.i686}
+                elif native_architecture == Architecture.AMD64:
+                    return {Architecture.x86}
+                else:
+                    return set()
+        elif native_architecture in {Architecture.i686, Architecture.x86}:
+            return {native_architecture} if bitiness == "32" else set()
+        else:
+            # assert_never doesn't work here, oddly, maybe due to set checking above
+            raise RuntimeError("Cannot be reached")
```

Comment:
This looks a little overcomplicated to me... maybe this would work?

```python
    archs_32 = {Architecture.i686, Architecture.x86}
    auto_archs = Architecture.auto_archs(platform)

    if bitness == '64':
        return auto_archs - archs_32
    elif bitness == '32':
        return auto_archs.intersection(archs_32)
    else:
        assert_never(bitness)
```

Reply:
Didn't think of that, very clean. :)